### PR TITLE
rsdk-3691 rtk-system module: serial base station

### PR DESCRIPTION
**Pull request by oliviamiller**
### Summary
Creates module for base station. Only serial base-station is implemented for now. Changed type to sensor and made config changes from scope doc.

### Testing
Verified running the module on app.viam works and is writing correction data.

### CODE REVIEW

File: correction-station/correction_station.go [hunk 1]
```diff
@@ -17,6 +17,18 @@ import (
...
 
+const (
+	i2cStr    = "i2c"
+	serialStr = "serial"
+	timeMode  = "time"
+)
+
+var (
+	StationModel         = resource.NewModel("viam-labs", "sensor", "correction-station")
+	errStationValidation = fmt.Errorf("only serial, I2C are supported for %s", StationModel.Name)
```
Comment (randhid) [inline correction-station/correction_station.go]:
```suggestion
	errStationValidation = fmt.Errorf("only serial, i2c are supported for %s", StationModel.Name)
```

File: correction-station/correction_station.go [hunk 2]
```diff
@@ -17,6 +17,18 @@ import (
...
 )
 
+const (
+	i2cStr    = "i2c"
+	serialStr = "serial"
+	timeMode  = "time"
+)
+
+var (
+	StationModel         = resource.NewModel("viam-labs", "sensor", "correction-station")
```
Comment (randhid) [inline correction-station/correction_station.go:27]: does this need to be exported?

**Review by randhid — APPROVED** [review]

File: correction-station/correction_station.go [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+	}
+
+	r.logger.Debug("Starting")
+
+	r.Start(ctx)
+	return r, r.err.Get()
+}
+
+// Start starts reading from the correction source and sends corrections to the radio/bluetooth.
+func (r *rtkStation) Start(ctx context.Context) {
```
Comment (randhid) [inline correction-station/correction_station.go]: try to make most of the functions that are driver specific unexported as you go along. Not in this pr but as you edit this module further.

File: correction-station/correction_station.go (hunk 2 above)
Comment (oliviamiller) [reply correction-station/correction_station.go:27]: yes since its being used in main.go

File: correction-station/correction_station.go [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+// I2CConfig is used for converting attributes for a correction source.
+type I2CConfig struct {
+	Board       string `json:"board"`
+	I2CBus      string `json:"i2c_bus"`
+	I2cAddr     int    `json:"i2c_addr"`
+	I2CBaudRate int    `json:"i2c_baud_rate,omitempty"`
+}
+
+// Validate ensures all parts of the config are valid.
+func (cfg *StationConfig) Validate(path string) ([]string, error) {
```
Comment (randhid) [inline correction-station/correction_station.go]: Eventually, we can break up the no network GPS rtk into a different package in this module, so you can keep the StationCofnig here and rename it Config. Resource is looking for an object called Config with a Validate function for each registered resource model.

**Review by randhid — APPROVED** [review]
Nice! I'm sure we'll keep improving this code as we go along!

**Review by susWorld — APPROVED** [review]
🖖 Looks good for now. I am expecting more polishing as this stabilize. Well done!

**Pull request merged**